### DISCUSSION TITLE
removing all textures

### DISCUSSION THREAD

**Discussion by ixxie** [donmccurdy/glTF-Transform#806] [Q&A] [answered]
My project doesn't require textures at all; how does one completely remove all textures?

**Comment by donmccurdy** [comment, marked as answer]
Hi! This isn't currently a feature of the commandline, are you OK with using the node.js or web-based scripting APIs? If so you'd:

1. [choose an I/O service](https://gltf-transform.donmccurdy.com/classes/core.platformio.html)
2. load the glTF document

```javascript
import { NodeIO } from '@gltf-transform/core';
import { ALL_EXTENSIONS } from '@gltf-transform/extensions';

const io = new NodeIO().registerExtensions(ALL_EXTENSIONS);
const document = await io.read('path/to/scene.glb');
```

3. iterate over textures and dispose them

```javascript
for (const texture of document.getRoot().listTextures()) {
  texture.dispose();
}
```

4. write the glTF document

```javascript
await io.write('path/to/scene.glb', document);
```

You could also remove materials in the same way, if you want, or use `dedup()` to combine any materials that are no longer distinct after textures have been removed:

```javascript
import { dedup } from '@gltf-transform/functions';

await document.transform( dedup() );
```

**Reply by donmccurdy** [reply]
In context of https://github.com/nytimes/rd-bundler-3d-plugins/issues/14, if you'd like to wrap all of that up and use it with a vite or rollup plugin, the plugin transform list would look something like this:

```javascript
// vite.config.js
import gltf from 'vite-plugin-gltf';
import { dedup } from '@gltf-transform/functions';

function removeTextures() {
  return (document) => {
    for (const texture of document.getRoot().listTextures()) {
      texture.dispose();
    }
  }
}

return {
  // ...
  plugins: [
    gltf({ 
      transforms: [ removeTextures(), dedup() ] 
    })
  ]
}
```
